pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -145,6 +145,9 @@ def on_run_end(self) -> List[_OUT_DICT]:
         # hook
         self._on_evaluation_epoch_end()
 
+        # hook
+        self._on_evaluation_end()
```

Comment:
just concerned with an edge case where a user logs inside this hook and expect `ModelCheckpoint` or `EarlyStopping` to monitor it, or some other crazy callback using `callback_metrics/logged_metrics`.

where do we draw a line and have a hook where we expect all the logs will be present?

Reply:
This is not a problem after the logger redesign because `ModelCheckpoint` saves (for validation) also during `on_validation_end` and it's the last callback to run.

Reply:
but we run callbacks before lightningmodule:
https://github.com/PyTorchLightning/pytorch-lightning/blob/1a2536361749d13e3f30cec11c7da11747f425ab/pytorch_lightning/loops/dataloader/evaluation_loop.py#L229-L231

Reply:
Mmm great point.

Yeah then this won't work well for the `LightningModule`.

So we have these options:
1. Drop this feature
1. Go forward knowing this
1. Invert the order for this hook in particular (not a great idea as it would be a breaking change).

What do you think is the best? We should probably mention this in the issue.

Reply:
personally 1. since we have a workaround of using `on_validation_epoch_end`. Also, the user seems to be happy with it: https://github.com/PyTorchLightning/pytorch-lightning/issues/5285#issuecomment-1017022560

2. IMO, will leak an issue even though many users maybe not be using it, but we can't tell for sure.
3. same reason.